src/main/java/com/exasol/adapter/AdapterProperties.java

```diff
@@ -180,21 +185,138 @@ public static boolean isRefreshingVirtualSchemaRequired(final Map<String, String
+     * Check if the data source is the local Exasol instance
      *
-     * @return empty map
+     * @return <code>true</code> if the data source and the database containing the
+     *         Virtual Schema are hosted on the same Exasol cluster
      */
-    public static AdapterProperties emptyProperties() {
-        return new AdapterProperties(Collections.emptyMap());
+    public boolean isLocalSource() {
+        return isEnabled(IS_LOCAL_PROPERTY);
     }
 
     /**
-     * Check if the data source is the local Exasol instance
+     * Check if the username property is set
      *
-     * @return <code>true</code> if the data source and the database containing the Virtual Schema are hosted on the
-     *         same Exasol cluster
+     * @return <code>true</code> if username property is set
      */
-    public boolean isLocalSource() {
-        return isEnabled(IS_LOCAL_PROPERTY);
+    public boolean hasUsername() {
+        return containsKey(USERNAME_PROPERTY);
+    }
+
+    /**
+     * Check if the password property is set
+     *
+     * @return <code>true</code> if password property is set
+     */
+    public boolean hasPassword() {
+        return containsKey(PASSWORD_PROPERTY);
+    }
+
+    /**
+     * Check if the connection string property is set
+     *
+     * @return <code>true</code> if connection string property is set
+     */
+    public boolean hasConnectionString() {
+        return containsKey(CONNECTION_STRING_PROPERTY);
+    }
+
+    /**
+     * Check if the table filter property is set
+     *
+     * @return <code>true</code> if table filter property is set
+     */
+    public boolean hasTableFilter() {
+        return containsKey(TABLE_FILTER_PROPERTY);
+    }
+
+    /**
+     * Check if the catalog name property is set
+     *
+     * @return <code>true</code> if catalog name property is set
+     */
+    public boolean hasCatalogName() {
+        return containsKey(CATALOG_NAME_PROPERTY);
+    }
+
+    /**
+     * Check if the schema name property is set
+     *
+     * @return <code>true</code> if schema name property is set
+     */
+    public boolean hasSchemaName() {
+        return containsKey(SCHEMA_NAME_PROPERTY);
+    }
+
+    /**
+     * Check if the connection name property is set
+     *
+     * @return <code>true</code> if connection name property is set
+     */
+    public boolean hasConnectionName() {
+        return containsKey(CONNECTION_NAME_PROPERTY);
+    }
+
+    /**
+     * Check if the debug address property is set
+     *
+     * @return <code>true</code> if debug address property is set
+     */
+    public boolean hasDebugAddress() {
+        return containsKey(DEBUG_ADDRESS_PROPERTY);
+    }
+
+    /**
+     * Check if the log level property is set
+     *
+     * @return <code>true</code> if log level property is set
+     */
+    public boolean hasLogLevel() {
+        return containsKey(LOG_LEVEL_PROPERTY);
+    }
+
+    /**
+     * Check if the sql dialect property is set
```

Comment:
`     * Check if the SQL dialect property is set`

Reply:
fixed